sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -47,16 +46,16 @@ import org.apache.spark.util.ArrayImplicits._
  * Note: [[HDFSMetadataLog]] doesn't support S3-like file systems as they don't guarantee listing
  * files in a directory always shows the latest files.
  */
-class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path: String)
-  extends MetadataLog[T] with Logging {
-
-  private implicit val formats = Serialization.formats(NoTypeHints)
+class HDFSMetadataLog[T <: AnyRef: ClassTag](sparkSession: SparkSession, path: String)(
+    private final implicit val manifest: Manifest[T])
```

Comment:
why we need add this implicit definition?

Reply:
for  class `CompactibleFileStreamLog` and `HDFSMetadataLog`, they has `T`, and implement method `serialize` ,`deserialize`. And `deserialize` use `org.json4s.jackson.JacksonSerialization#read`
`def read[A](in: Reader)(implicit formats: Formats, mf: Manifest[A]): A = ...`

if we use `private implicit val manifest = Manifest.classType[T](implicitly[ClassTag[T]].runtimeClass)`,
the manifest can get T from ClassTag by type inference.

if we use `private implicit val manifest : Manifest[T] = Manifest.classType[T](implicitly[ClassTag[T]].runtimeClass)`
the T is will be Object, then manifest can't get `T`.

We should declare Maintest [T] in the constructor signature, or use a variable to save Classtag and use Classtag at runtime.

Reply:
talk is cheap, show code :
![image](https://github.com/apache/spark/assets/35491928/9bdb9595-a97f-4db1-96bc-ce48305aa33c)
![image](https://github.com/apache/spark/assets/35491928/36faf445-3731-4bb1-90d3-3571cdc167e7)
![image](https://github.com/apache/spark/assets/35491928/72870aa8-6a6f-4c76-9a3e-f2cc6412602c)

manifest is used to deserialize for Json4s

Reply:
done